### DISCUSSION TITLE
Refresh table with new value

### DISCUSSION THREAD

**Discussion by laserir** [zauberzeug/nicegui#1705] [Q&A] [answered]
### Question

Having seen this solution (https://github.com/zauberzeug/nicegui/discussions/1307), it seems not quite to be working for me.

The table shows all data as intended when set up like 
`err_table = ui.table(columns=columns,rows=get_all_error_messages(o.data))`.
There is a timer running every secound that should update the table with new values returned by get_all_error_messges: `err_table.update()` or `ui.update()` dont throw exceptions, but also dont refresh the table.

Any ideas? :-O

**Comment by falkoschindler** [comment]
Can you, please, provide a [minimal reproducible code example](https://en.wikipedia.org/wiki/Minimal_reproducible_example). This allows us to help more efficiently. Thanks!

**Reply by laserir** [reply]
Will do asap!

**Comment by natankeddem** [comment]
`err_table` won't update if you call `hardware_table.update()` you need to call `err_table.update()`

**Reply by laserir** [reply]
Sorry, of course it was both the same table.

**Comment by laserir** [comment]
```py
import ...
s=None
o=None
err_table=None

class Operation:
	def __init__(self, settings) -> None:
		self.settings = settings
		self.data = {"system": {},"hardware":{}, "error": []}
		for system_key, system_value in self.settings.data["system"].items():
			self.data["system"][system_key] = {"id": system_key}

	def update(self) -> None:
		global err_table
		if err_table is not None:      #Avoid error while table is not initialized
			err_table.update()
			print("Debug: Updated")
			print(o.data)

def get_all_error_messages(errors_data):
    all_errors = []

    for system in errors_data.get("error", []):
        system_id = system.get("system_id")
        for error_message in system.get("error_messages", []):
            error = {
                "system_id": system_id,
            }
            all_errors.append(error)

    return all_errors

@ui.page('/')
def main_page() -> None:
    global err_table
    with ui.tab_panels(tabs):
        with ui.tab_panel('Error'):
            columns = [
                {'name': 'system_id', 'label': 'system_id', 'field': 'system_id', 'sortable': True, 'align': 'left'},
            ]

            err_table = ui.table(columns=columns,rows=get_all_error_messages(o.data))

s = Settings(load_json(config_file))
o = Operation(s)
ui.run()
```

o.update() is called every second; debug message prints fine. o.data is (succeesfully) changed by a different threat; debug message shows the data correct, but table doesnt get updated.

**Reply by falkoschindler** [reply]
This code is neither minimal (e.g. the tabs are probably not needed) nor executable/reproducible (e.g. `Settings` is undefined).

**Comment by natankeddem** [comment]
In the original discussion you linked to `rows` is set to a list which is then able to be modified and is linked to the rows value in the `table` when you call `update`. Essentially in the example you are passing a reference to the `rows` list, because that is how python works, a new list is not created. 

Here you are getting rows once on your initial instantiation of `table` via `get_all_error_messages` then you are doing nothing else to manipulate the row data. You can do the same thing as the example and set `rows` to a list reference. The alternative is that you can manipulate the `table` via the methods `add_rows` or `remove_rows`.

**Comment by laserir** [comment, marked as answer]
As the row list is changing completely, I solved it like this:
```
        # Update Error Table
        global err_table
        if err_table is not None:    # Prevent error before err_table is initialized
            err_table.rows[:] = self.get_errors(readable=True)
            err_table.update()
```
